fix(core): Improve DebugElement.query() return type

JeanMeche opened the pull request:
`DebugElement.query()` can return null, let's reflect that on the return type. This commit also provides a migration to insert non-null assertions after `DebugElement.query()` on existing code bases.

Fixes #22449

## PR Type
What kind of change does this PR introduce?

- [x] Feature


## Does this PR introduce a breaking change?

- [x] Yes, should target the next major.

JeanMeche commented:
How can I run the new migration schematic on the angular project itself ?

atscott commented:
For reference, there was previous conversation around whether it would be better to throw an error rather than return `null`. https://github.com/angular/angular/pull/39632#issuecomment-1031774045 Returning `null` aligns with `document.querySelector` and has an easier migration. 

I think this should be marked as a `fix` rather than a feature.

atscott reviewed (approved):
reviewed-for: public-api

Reviewed only for the public API change of adding `null` to the type.

JeanMeche committed:
fix(core): Improve DebugElement.query() return type

`DebugElement.query()` can return null, let's reflect that on the return type.
This commit also provides a migration to insert non-null assertions after `DebugElement.query()` on existing code bases.

Fixes #22449

JeanMeche commented:
I'm going through my old PR, I believe this one will shift to at least v17, right ?

AndrewKushnir commented:
> I'm going through my old PR, I believe this one will shift to at least v17, right ?

Yes, I've added it to the `v17-candidates` milestone.
